source/org/zfin/mapping/repository/HibernateLinkageRepository.java

```diff
@@ -160,17 +157,27 @@ public List<Marker> getMappedClonesContainingGene(Marker marker) {
     @Override
     public List<MappedMarker> getMappedMarkers(Panel panel, ZdbID marker, String lg) {
 
-        Criteria criteria = HibernateUtil.currentSession().createCriteria(MappedMarker.class);
+        String hql = "from MappedMarker ";
+        List<String> hqlClauses = new ArrayList<>();
+        HashMap<String, Object> parameterMap = new HashMap<>();
+
+
         if (lg != null) {
-            criteria.add(Restrictions.eq("lg", lg));
+            hqlClauses.add("lg = :lg");
+            parameterMap.put("lg", lg);
         }
         if (marker != null) {
-            criteria.add(Restrictions.eq("marker", marker));
+            hqlClauses.add("entityID = :entityID");
```

Comment:
I saw that this entityID is a new field on MappedMarker. I assume it's all populated correctly.

Reply:
Yes, that is working (also see unit test). I think this was a bug before.